src/go/rpk/pkg/cli/topic/describe_storage.go

```diff
@@ -0,0 +1,258 @@
+				}
+				// We can safely pick the value of the first partition since
+				// the cloud storage mode doesn't change between partitions of
+				// the same topic.
+				tw.PrintColumn("CLOUD-STORAGE-MODE", report[0].CloudStatus.CloudStorageMode)
+
+				lastUpload := getLastUpload(report)
+				tw.PrintColumn("LAST-UPLOAD", humanReadable(lastUpload, human, "time"))
+			})
+
+			header("LOG-SIZE", size, withSections, func() {
+				tw := out.NewTable("PARTITION", "TOTAL", "CLOUD", "LOCAL")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(
+						r.Partition,
+						humanReadable(r.CloudStatus.TotalLogSize, human, "size"),
+						humanReadable(r.CloudStatus.CloudLogSize, human, "size"),
+						humanReadable(r.CloudStatus.LocalLogSize, human, "size"),
+					)
+				}
+			})
+
+			header("SYNC", sync, withSections, func() {
+				isRrr := report[0].CloudStatus.CloudStorageMode == "read_replica"
+				headers := []string{"PARTITION", "LAST-SEGMENT-UPLOAD", "LAST-MANIFEST-UPLOAD", "METADATA-UPDATE-PENDING"}
+				if isRrr {
+					headers = append(headers, "LAST-MANIFEST-SYNC")
+				}
+				tw := out.NewTable(headers...)
+				defer tw.Flush()
+				for _, r := range report {
+					row := []any{
+						r.Partition,
+						humanReadable(r.CloudStatus.LastSegmentUpload, human, "time"),
+						humanReadable(r.CloudStatus.LastManifestUpload, human, "time"),
+						r.CloudStatus.MetadataUpdatePending,
+					}
+					// If isRrr is true, LastManifestSync should be there, this
+					// nil check is possibly redundant, but we want to avoid a
+					// nil pointer dereference.
+					if isRrr && r.CloudStatus.LastManifestSync != nil {
+						row = append(row, humanReadable(*r.CloudStatus.LastManifestSync, human, "time"))
+					}
+					tw.Print(row...)
+				}
+			})
+
+			header("LOG-SEGMENT-COUNT", count, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD", "LOCAL")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogSegmentCount, r.CloudStatus.LocalLogSegmentCount)
+				}
+			})
+
+			header("LOG-OFFSET", offset, withSections, func() {
+				tw := out.NewTable("PARTITION", "CLOUD-START", "CLOUD-LAST", "LOCAL-START", "LOCAL-LAST")
+				defer tw.Flush()
+				for _, r := range report {
+					tw.Print(r.Partition, r.CloudStatus.CloudLogStartOffset, r.CloudStatus.CloudLogLastOffset, r.CloudStatus.LocalLogStartOffset, r.CloudStatus.LocalLogLastOffset)
+				}
+			})
+		},
+	}
+	p.InstallAdminFlags(cmd)
+
+	cmd.Flags().BoolVarP(&all, "print-all", "a", false, "Print all cloud storage status")
+	cmd.Flags().BoolVarP(&summary, "print-summary", "s", false, "Print the summary section")
+	cmd.Flags().BoolVar(&size, "print-size", false, "Print the log size section")
+	cmd.Flags().BoolVar(&sync, "print-sync", false, "Print the sync section")
+	cmd.Flags().BoolVar(&count, "print-count", false, "Print the log count section")
+	cmd.Flags().BoolVar(&offset, "print-offset", false, "Print the offset section")
```

Comment:
maybe some short flag recommendations, in order:
-s (summary)
-z (size)
-y (sync)
-c (count)
-o (offset)
?
It's a bit cryptic but can help if you want these specific sections a lot

Reply:
`rpk debug bundle` uses -o for output file
`rpk topic produce` uses it for output format
`rpk topic consume` uses -o for offset to start at

So, we don't really have consistency at the moment

Reply:
I don't think we will ever have --output here, but --format, so I'm using -o.